src/validation.cpp

```diff
@@ -1975,6 +1976,16 @@ bool CChainState::ConnectBlock(const CBlock& block, BlockValidationState& state,
     int64_t nTime6 = GetTimeMicros(); nTimeCallbacks += nTime6 - nTime5;
     LogPrint(BCLog::BENCH, "    - Callbacks: %.2fms [%.2fs (%.2fms/blk)]\n", MILLI * (nTime6 - nTime5), nTimeCallbacks * MICRO, nTimeCallbacks * MILLI / nBlocksTotal);
 
+    TRACE7(validation, block_connected,
+        block.GetHash().ToString().c_str(),
+        pindex->nHeight,
+        block.vtx.size(),
+        nInputs,
+        nSigOpsCost,
+        GetTimeMicros() - nTimeStart, // in microseconds (µs)
```

Comment:
although this is convenient, it does use up a valuable argument slot. we could achieve the same thing with a uprobe:

```bpftrace
uprobe:bitcoind:"CChainState::ConnectBlock(CBlock const&, BlockValidationState&, CBlockIndex*, CCoinsViewCache&, CChainParams const&, bool)" 
{ 
    @connect_start = nsecs;
}

usdt:bitcoind:validation:block_connected
{
    @t = nsecs - @connect_start;
}
```

(untested)

alternatively if we want a more stable interface we could just stick a usdt at the start as well.

Reply:
I'm not sure if we want to add two tracepoints (one fn start, one fn end) for measuring function timing. This would also require us to pass and handle some kind of identifier for functions running in parallel (not the case for `ConnectBlock`).